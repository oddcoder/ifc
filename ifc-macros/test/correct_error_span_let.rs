use ifc_macros::ifc_block;


fn main() {
    ifc_block! {
        #[IFC(Low)]
        let w : u64 = 1;
        let x : u64 = 2;
        #[IFC(Low)]
        let z : u64 = w * x;
        println!("z = {}", z);
    };
}
