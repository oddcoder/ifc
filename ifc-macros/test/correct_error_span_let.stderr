error: Cannot assign high expression to low variable
  --> test/correct_error_span_let.rs:10:9
   |
10 |         let z : u64 = w * x;
   |         ^^^^^^^^^^^^^^^^^^^^
   |
help: Low Variable
  --> test/correct_error_span_let.rs:10:13
   |
10 |         let z : u64 = w * x;
   |             ^
help: High Expression
  --> test/correct_error_span_let.rs:10:23
   |
10 |         let z : u64 = w * x;
   |                       ^^^^^
